machineconfiguration/v1alpha1/types_machineosconfig.go

```diff
@@ -0,0 +1,199 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=machineosconfigs,scope=Cluster
+// +kubebuilder:subresource:status
+// +openshift:api-approved.openshift.io=https://github.com/openshift/api/pull/1773
+// +openshift:enable:FeatureGate=OnClusterBuild
+// +openshift:file-pattern=cvoRunLevel=0000_80,operatorName=machine-config,operatorOrdering=01
+// +kubebuilder:metadata:labels=openshift.io/operator-managed=
+
+// MachineOSConfig describes the configuration for a build process managed by the MCO
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type MachineOSConfig struct {
```

Comment:
I keep reading this and it feels like it should be `MachineOsBuildConfig`?

Reply:
ehhh, I think MachineOSConfig is good. it is the config for building the OS, we will probably over time add more options here that lean towards configuring the OS rather than the build process, naming it `MachinOSBuildConfig` will close that door and create the opposite issue. 

WDYT @cheesesashimi ?

Reply:
Ok but logically you have two separate things here or maybe I  am misunderstanding.
- a build with a config
- an os image which is part of the build which also has a config

Reply:
So given above would the build process itself ever need its own configurations? In my mind its like this.
```
build:
  - config:
    - image:
       - name: 
       - config: 
    - other_build_params:
```

Reply:
your thinking of the structure makes sense to me,  I was just saying I'd like to keep the name open, given that we have a roadmap of different spec entities we are going to add to MachineOSConfig. I don't want to tie it too much to the name "Build". That is what the `MachineOSBuild` is for :)